controllers/disruption_schedule_controller.go

```diff
@@ -0,0 +1,221 @@
+	r.log.Info("Reconciling DisruptionSchedule")
+
+	instance := &chaosv1beta1.DisruptionSchedule{}
+
+	// retrieve schedule
+	if err := r.Client.Get(context.Background(), req.NamespacedName, instance); err != nil {
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	}
+
+	if !instance.DeletionTimestamp.IsZero() {
+		// NOTE: add a finalizer if anything needs to be written here
+		// The DisruptionSchedule is being deleted.
+		return ctrl.Result{}, nil
+	}
+
+	disruptions, err := r.getChildDisruptions(ctx, instance)
+	if err != nil {
+		return ctrl.Result{}, nil
+	}
+
+	if err := r.updateLastScheduleTime(ctx, instance, disruptions); err != nil {
+		r.log.Errorw("unable to update LastScheduleTime of DisruptionSchedule status", "DisruptionSchedule", instance.Name, "err", err)
+		return ctrl.Result{}, err
+	}
+
+	// _ is targetResourceNotFound, which will be used later to requeue according to the schedule
+	_, instanceDeleted, err := r.updateTargetResourcePreviouslyMissing(ctx, instance)
+	if err != nil {
+		// Error occurred during status update or deletion, requeue
+		r.log.Errorw("failed to handle target resource status", "DisruptionSchedule", instance.Name, "err", err)
+		return ctrl.Result{}, err
+	}
+
+	if instanceDeleted {
+		// The instance has been deleted, reconciliation can be skipped
+		return ctrl.Result{}, nil
+	}
+
+	return ctrl.Result{}, nil
+}
+
+func (r *DisruptionScheduleReconciler) getChildDisruptions(ctx context.Context, instance *chaosv1beta1.DisruptionSchedule) (*chaosv1beta1.DisruptionList, error) {
+	disruptions := &chaosv1beta1.DisruptionList{}
+	labelSelector := labels.SelectorFromSet(labels.Set{DisruptionScheduleNameLabel: instance.Name})
+
+	if err := r.Client.List(ctx, disruptions, client.InNamespace(instance.Namespace), &client.ListOptions{LabelSelector: labelSelector}); err != nil {
+		r.log.Errorw("unable to list Disruptions", "DisruptionSchedule", instance.Name, "err", err)
+		return disruptions, err
+	}
+
+	return disruptions, nil
+}
+
+// updateLastScheduleTime updates the LastScheduleTime in the status of a DisruptionSchedule instance based on the most recent schedule time among the given disruptions
+func (r *DisruptionScheduleReconciler) updateLastScheduleTime(ctx context.Context, instance *chaosv1beta1.DisruptionSchedule, disruptions *chaosv1beta1.DisruptionList) error {
+	mostRecentScheduleTime := r.getMostRecentScheduleTime(disruptions) // find the last run so we can update the status
+	if mostRecentScheduleTime != nil {
+		instance.Status.LastScheduleTime = &metav1.Time{Time: *mostRecentScheduleTime}
+	} else {
+		instance.Status.LastScheduleTime = nil
+	}
+
+	return r.Client.Status().Update(ctx, instance)
+}
+
+// getMostRecentScheduleTime returns the pointer to the most recent scheduled time among a list of disruptions
+func (r *DisruptionScheduleReconciler) getMostRecentScheduleTime(disruptions *chaosv1beta1.DisruptionList) *time.Time {
```

Comment:
This approach is very unlikely to work in most cases. We'll only be able to `Get` existing disruptions into this list, but disruptions do expire and get garbage collected, so we'll very frequently (almost always) just have an empty list here. If I understand the Status field we're trying to populate with this calculation, it won't be useful to constantly be setting it to 0/nil. We'll only be able to know the most recently success of a disruption if we have just created it and are following its progress

Reply:
I think I might need to keep it. If we scheduled the disruption, but failed to update the LastScheduleTime on creation, we need to requeue so that the status could be updated. In this scenario at least one the disruption that we have scheduled will be available for us to update the status.